arch/x86/kvm/pkvm/pkvm.c

```diff
@@ -226,6 +228,166 @@ static void pkvm_vm_destroy(int vm_handle, struct pkvm_memcache *mc)
 	donate_mem_in_memcache(mc);
 }
 
+static int attach_pkvm_vcpu_to_vm(struct pkvm_vm *pkvm_vm, struct pkvm_vcpu *pkvm_vcpu)
+{
+	struct kvm *kvm = &pkvm_vm->kvm;
+	int vcpu_handle;
+
+	pkvm_spin_lock(&pkvm_vm->lock);
+
+	if (kvm->created_vcpus == KVM_MAX_VCPUS) {
+		pkvm_spin_unlock(&pkvm_vm->lock);
+		return -EINVAL;
+	}
+	vcpu_handle = kvm->created_vcpus++;
+	pkvm_vcpu->vcpu.arch.pkvm_vcpu_handle = vcpu_handle;
+	pkvm_vm->vcpus[vcpu_handle] = pkvm_vcpu;
+
+	pkvm_spin_unlock(&pkvm_vm->lock);
+
+	atomic_set(&pkvm_vm->vcpu_refs[vcpu_handle], 1);
+
+	return vcpu_handle;
+}
+
+static int pkvm_arch_vcpu_create(struct pkvm_vcpu *pkvm_vcpu, struct fpstate *fps)
```

Comment:
nit: just realized that "arch" in the function name may be confusing. What does it have to do with arch?

Maybe e.g. just `__vcpu_create()`?

(At first I wanted to suggest `__pkvm_vcpu_create()`, but it may be inconsistent, since we have `__pkvm_vcpu_free()` which does more stuff: detach, unshare..., while pkvm_arch_vcpu_create() just does the vcpu setup)

Reply:
Seems I don't have any better name than `__vcpu_create()`. Let's use it.

>(At first I wanted to suggest __pkvm_vcpu_create(), but it may be inconsistent, since we have __pkvm_vcpu_free() which does more stuff: detach, unshare..., while pkvm_arch_vcpu_create() just does the vcpu setup)

Yeah, the reason to have `__pkvm_vcpu_free()` is to make it shareable with `pkvm_vm_destroy()`